### PR TITLE
JWK support for RSAKey

### PR DESCRIPTION
**Pull request by valeriomazzeo**
This pull request adds JSON Web Key support for RSAKey.

### CODE REVIEW

File: Sources/JWT/Signers/RSAKey+JWK [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+    public init(n: String, e: String, d: String? = nil) throws {
+
+        func parseBignum(_ s: String) -> UnsafeMutablePointer<BIGNUM> {
+            return s.makeBytes().base64URLDecoded.withUnsafeBufferPointer { p in
+                return BN_bin2bn(p.baseAddress, Int32(p.count), nil)
+            }
+        }
+
+        let rsa = RSA_new()!
```
Comment (siemensikkema) [inline Sources/JWT/Signers/RSAKey+JWK]: Prefer `guard let rsa = ...` over force unwrapping. You can throw `JWTError.createKey`.

File: Sources/JWT/Signers/RSAKey+JWK [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import Foundation
+import CTLS
+
+public extension RSAKey {
+
+    public init(n: String, e: String, d: String? = nil) throws {
```
Comment (siemensikkema) [inline Sources/JWT/Signers/RSAKey+JWK]: Can you document the input parameters?

File: Sources/JWT/Signers/RSAKey+JWK [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        rsa.pointee.e = parseBignum(e)
+
+        if let d = d {
+            rsa.pointee.d = parseBignum(d)
+            self = .private(rsa)
+        } else {
+            self = .public(rsa)
+        }
+    }
+}
```
Comment (siemensikkema) [inline Sources/JWT/Signers/RSAKey+JWK]: Missing newline

**Comment by valeriomazzeo** [comment]
@siemensikkema done 👍

**Comment by siemensikkema** [comment]
@tanner0101 can you merge this? If this and #61 get merged then https://github.com/vapor/jwt-provider/pull/14 can be finalized as well.

**Comment by valeriomazzeo** [comment]
@tanner0101 @LoganWright is there any chance we can go ahead with this?

**Pull request merged**